src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -1228,8 +1230,16 @@ protected long getApproxSizeOfRecordInBytes(SinkRecord kafkaSinkRecord) {
           }
         }
       }
-    } catch (JsonProcessingException e) {
-      // We ignore any errors here because this is just calculating the record size
+    } catch (Exception e) {
+      boolean isJsonProcessingEx = e instanceof JsonProcessingException;
+      boolean isSnowflakeParsingEx =
+          e instanceof SnowflakeKafkaConnectorException
+              && ((SnowflakeKafkaConnectorException) e).checkErrorCode(SnowflakeErrors.ERROR_0010);
+      if (!(isJsonProcessingEx || isSnowflakeParsingEx)) {
+        // We ignore any errors parsing exceptions here because this is just calculating the record
+        // size
+        throw new RuntimeException(e);
```

Comment:
Why we're not throwing the original exception?

Reply:
`JsonProcessingException` is checked exception so the compiler requires to add throws clause to the signature of `getApproxSizeOfRecordInBytes`. The call stack leads to `SnowflakeSinkTask#put()` which is Kafka Connect method that we cannot modify. At some point we have to wrap checked exception into the unchecked one.